Centaurus.Exchange.Analytics/Frames/OHLCFrame.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Centaurus.Analytics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Centaurus.Exchange.Analytics
+{
+    public class OHLCFrame : IEquatable<OHLCFrame>
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="startTime">Trimmed date time</param>
+        /// <param name="period"></param>
+        public OHLCFrame(DateTime startTime, OHLCFramePeriod period, int market)
+        {
+            StartTime = startTime;
+            Period = period;
+            Market = market;
+        }
+
+        public DateTime StartTime { get; }
+
+        public OHLCFramePeriod Period { get; }
+
+        public int Market { get; }
+
+        public double Hi { get; set; }
+
+        public double Low { get; set; }
+
+        public double Open { get; set; }
+
+        public double Close { get; set; }
+
+        public double Volume { get; set; }
+
+        public void OnTrade(Trade trade)
+        {
+            if (trade == null)
+                throw new ArgumentNullException(nameof(trade));
+            if (Open == default) //register first trade
+            {
+                Open = Hi = Low = Close = trade.Price;
+                Volume = trade.BaseAmount;
+                return;
+            }
+            if (Hi < trade.Price)
+                Hi = trade.Price;
+            if (Low > trade.Price)
+                Low = trade.Price;
+            Close = trade.Price;
+            Volume += trade.BaseAmount;
+        }
+
+        public bool IsExpired(DateTime currentDateTime)
+        {
+            return StartTime.GetDiff(currentDateTime, Period) != 0;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as OHLCFrame);
+        }
+
+        public bool Equals(OHLCFrame other)
```

Comment:
Where do you use those comparisons? Looks like it can be simplified down to comparing StartTime, Market, and Period. Or, if you create the generic Id encoding, you'll be able to compare straight by ids, and it's much better option.